tests/integration/src/test/scala/mesosphere/marathon/integration/DockerAppIntegrationTest.scala

```diff
@@ -2,18 +2,39 @@ package mesosphere.marathon
 package integration
 
 import mesosphere.marathon.integration.facades.MarathonFacade._
-import mesosphere.marathon.integration.setup.{EmbeddedMarathonTest, MesosConfig}
+import mesosphere.marathon.integration.setup.{EmbeddedMarathonTest, MesosConfig, ProcessOutputToLogStream}
 import mesosphere.marathon.raml.{App, Container, DockerContainer, EngineType, Network, NetworkMode}
 import mesosphere.marathon.state.PathId._
 import mesosphere.{AkkaIntegrationTest, WhenEnvSet}
 
+import scala.sys.process.Process
+
 class DockerAppIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathonTest {
 
   override lazy val mesosConfig = MesosConfig(containerizers = "docker,mesos")
 
   // FIXME (gkleiman): Docker tests don't work under Docker Machine yet. So they can be disabled through an env variable.
   val envVar = "RUN_DOCKER_INTEGRATION_TESTS"
 
+  def waitForDocker(retries: Int): Unit = {
+    val timeoutSeconds = 5 // Mesos agent waits 5 seconds
+    val p = Process(command = Seq("timeout", (timeoutSeconds * 1000).toString, "docker", "-H", "/var/run/docker.sock", "--version"))
```

Comment:
Fun fact: we already do this before all tests in [ci/pipeline](https://github.com/mesosphere/marathon/blob/master/ci/pipeline#L170)